### PR TITLE
Add docs for heartbeat hints builder

### PR DESCRIPTION
**Pull request by vjsamuel**
Apologies for not doing this sooner. @andrewvc

### CODE REVIEW

**Comment by jsoriano** [comment]
I think Andrew already added some docs, @andrewvc would you like to take a look to this PR in case we can add something to the docs we have?

**Comment by vjsamuel** [comment]
@jsoriano using some of the examples in this PR might be useful especially while running in a kubernetes environment. if it makes sense, I can rebase and try to add content of my PR into the current docs. thoughts?

cc: @exekias

**Comment by jsoriano** [comment]
@vjsamuel this would be great, thanks!

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+In the above sample the processor definition tagged with `1` would be executed first.
+
+[float]
+==== Kubernetes
+
+Kubernetes autodiscover provider supports hints in Pod annotations. To enable it just set `hints.enabled`:
+
+[source,yaml]
+-----
+filebeat.autodiscover:
```
Comment (exekias) [inline heartbeat/docs/autodiscover-hints.asciidoc]: this should be hearbeat

File: heartbeat/docs/autodiscover-hints.asciidoc [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+
+[float]
+==== Docker
+
+Docker autodiscover provider supports hints in labels. To enable it just set `hints.enabled`:
+
+[source,yaml]
+-----
+filebeat.autodiscover:
```
Comment (exekias) [inline heartbeat/docs/autodiscover-hints.asciidoc]: same here

**Comment by dedemorton** [comment]
bump. This has been open for awhile. Do you plan to finalize the changes, or should I take this over?

**Comment by vjsamuel** [comment]
@dedemorton i can clean this up and submit it by end of week.

**Comment by dedemorton** [comment]
@vjsamuel Any progress? I know that these are extraordinary times. Let me know if I can do anything to help you move this forward. Thanks!

**Comment by vjsamuel** [comment]
@dedemorton apologies. i will have this rebased by tomorrow

**Comment by vjsamuel** [comment]
@dedemorton thanks for the patience. i have rebased and added in the docs. let me know if more of the old documentation could be reused.

**Comment by dedemorton** [comment]
The doc changes LGTM from an editorial perspective.

@exekias or @jsoriano Can you approve the technical accuracy of the changes? Thanks!

**Comment by dedemorton** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by dedemorton** [comment]
@jenkins test this again

**Comment by dedemorton** [comment]
@jenkins test this

**Comment by dedemorton** [comment]
jenkins run the tests please

**Comment by dedemorton** [comment]
jenkins run the tests please

**Comment by andrewvc** [comment]
Thanks for bringing this one into the repo everyone! It is much appreciated.

**Comment by dedemorton** [comment]
I went ahead and backported this change to get it into versioned branches. Thanks!

**Pull request merged**